firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/FirebaseModelDownloader.java

```diff
@@ -84,7 +94,7 @@ public static FirebaseModelDownloader getInstance(@NonNull FirebaseApp app) {
   /** @return The set of all models that are downloaded to this device. */
   @NonNull
   public Task<Set<CustomModel>> listDownloadedModels() {
-    throw new UnsupportedOperationException("Not yet implemented.");
+    return Tasks.forResult(sharedPreferencesUtil.listDownloadedModels());
```

Comment:
Since sharedPreferences are sync, you could try to create an actual task here to get concurrency.

Reply:
Done - thanks.